rpcserver.go

```diff
@@ -500,7 +500,11 @@ func handleAddNode(s *rpcServer, cmd interface{}, closeChan <-chan struct{}) (in
 	case "add":
 		err = s.server.ConnectNode(addr, true)
 	case "remove":
-		err = s.server.RemoveNodeByAddr(addr)
+		rerr := s.server.RemoveNodeByAddr(addr)
+		// This connection is still pending, cancel it.
+		if rerr != nil {
+			err = s.server.CancelPendingConnection(addr)
```

Comment:
I think it's probably better to do this entirely within `server.go` since then you don't have to worry about the internal implementation of the connmanager/server and the semantics of the API are clearly defined (`RemoveNodeByAddr` meaning "remove and cancel any outstanding connections").

Iff you choose to implement this here, then IMO you should define the "peer not found" error as a variable and check for it explicitly.